### ISSUE TITLE
Git based catalog not showing updated version available

### ISSUE THREAD

**Issue by riaan53** [rancher/rancher#16636]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1. Add a git based catalog
2. Install catalog item
3. Push a new version of the catalog using ranchers versioning scheme

**Result:**
Expected the installed catalog item to show that theres a new version available. I have waited more than 10mins and refreshed the paged and nothing happened.

Current view:
<img width="675" alt="screen shot 2018-11-18 at 14 20 53" src="https://user-images.githubusercontent.com/7716342/48672360-90393200-eb3d-11e8-9c2a-d79c2fc9e48f.png">

Git repo:
<img width="1077" alt="screen shot 2018-11-18 at 14 22 25" src="https://user-images.githubusercontent.com/7716342/48672352-7dbef880-eb3d-11e8-832a-f117e3559313.png">

Updated Chart.yaml
<img width="651" alt="screen shot 2018-11-18 at 14 24 01" src="https://user-images.githubusercontent.com/7716342/48672367-a810b600-eb3d-11e8-9200-76741aa16f5e.png">

**Other details that may be helpful:**
Repo used for testing https://gitlab.com/riaan53/charts 

Rancher logs:
```
W1118 12:22:18.010548 6 reflector.go:341] github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:144: watch of *v1.Event ended with: The resourceVersion for the provided watch is too old.
2018/11/18 12:27:04 [ERROR] CatalogController gitlab [catalog] failed with : Timeout in HTTP GET to [https://gitlab.com/riaan53/charts/index.yaml], did not respond in 30s
2018/11/18 12:27:04 [ERROR] CatalogController library [catalog] failed with : Timeout in HTTP GET to [https://git.rancher.io/charts/index.yaml], did not respond in 30s
 Wrap lines
```
It does not look like rancher detected it to be a git based catalog and treats it as a helm one.

<img width="1329" alt="screen shot 2018-11-18 at 14 44 07" src="https://user-images.githubusercontent.com/7716342/48672591-78af7880-eb40-11e8-802a-b300ca63540d.png">

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.1.1
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): RKE created with rancher helm installed (imported)
- Machine type (cloud/VM/metal) and specifications (CPU/memory): 3GB memory 1 CPU
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:46:00Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"clean", BuildDate:"2018-10-05T16:36:14Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker (use `docker info`):

```
Containers: 50
 Running: 36
 Paused: 0
 Stopped: 14
Images: 23
Server Version: 18.06.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 468a545b9edcd5932818eb9de8e72413e616e86e
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.14.73-rancher
Operating System: RancherOS v1.4.2
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 2.365GiB
Name: lucid-bardeen
ID: C3J7:MM77:M3FZ:FX5Z:SYUQ:JMAC:CZZ2:XBEP:IN33:25LM:IX5R:FKU3
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 127.0.0.0/8
Registry Mirrors:
 http://172.30.1.53:5000/
Live Restore Enabled: false
```

**Comment by superseb** [comment]
Timeout means it can't connect to that address anymore. When did you add it the first time and deployed an app from it and when did you push a new version? Doesn't look like the pod with the `rancher/rancher` container has internet connectivity. It falls back to HTTP get when `git clone` fails, probably also with a timeout. What happens when you run the command inside the container?

**Comment by riaan53** [comment]
Hi @superseb ,  I noticed this a while back and never got to raising the ticket.  The timeout issues might be something else. Question about that: If it falls back to https helm catalog due to timeout will it try the git version again in future?

I added the repo, installed the app and then immediately pushed a new version to gitlab. 

I tried it again now on fresh cluster. I do not see any timeouts now, git clone and internet works fine in the rancher container but the updated catalog app version is never displayed.

How I can get the catalog item to show a new version is to go into the page where I can launch new catalogs, and click on the refresh button a few times. The go to the installed catalogs and refresh the page a few times it will show up with a new version available.

Let me know if I can try something else.

<img width="723" alt="screen shot 2018-11-18 at 20 41 15" src="https://user-images.githubusercontent.com/7716342/48676701-1a4fbd80-eb73-11e8-9f1f-1b7b0fa25afa.png">
